### PR TITLE
Enable IPv6 dualstack support by default

### LINKED ISSUE #15003: Enable IPv6 dualstack support by default
### Description

In pull request: https://github.com/keycloak/keycloak/pull/13577 documentation was added on how to enable IPv6 in the JVM. I am running an EKS cluster in IPv6 mode and it was surprising to me that in 2022 a project would require additional configuration to support IPv6.

I would expect that Keycloak would work with both IPv4 and IPv6 by default and bind to both IPv4 and IPv6 on all interfaces by default.

### Discussion

_No response_

### Motivation

IPv6 on hosts where Keycloak is deployed is becoming far more mainstream and Keycloak should support receiving requests over IPv6.

### Details

Please enable IPv6 by default.

### PR DESCRIPTION
**Pull request by mabartos**
Closes #15003

@vmuzikar Could you please check it? Thanks

### CODE REVIEW

File: quarkus/tests/integration/src/test/java/org/keycloak/it/cli/dist/IpStackDistTest.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    @Launch({"start", "-Djava.net.preferIPv4Stack=true"})
+    public void ipv4Preferred() throws IOException {
+        assertThat(given().when().get("http://localhost:8080").getStatusCode(), is(200));
+        assertThat(given().when().get("http://127.0.0.1:8080").getStatusCode(), is(200));
+
+        try (CloseableHttpClient c = HttpClientBuilder.create().build(); Closeable ignore = c.execute(new HttpGet("http://[::1]:8080"))) {
+            Assertions.fail("Connection refused must be thrown");
+        } catch (HttpHostConnectException e) {
+            assertThat(ConnectException.class.equals(e.getCause().getClass()), is(true));
+        }
```
Comment (vmuzikar) [inline quarkus/tests/integration/src/test/java/org/keycloak/it/cli/dist/IpStackDistTest.java:61]: I think we can use something like this instead:
https://github.com/keycloak/keycloak/blob/72b238fb489aa9da257a5896772d6efde4a0f7d9/services/src/test/java/org/keycloak/vault/KeystoreVaultProviderTest.java#L67

**Review by vmuzikar — CHANGES_REQUESTED** [review]

File: docs/guides/server/outgoinghttp.adoc [hunk 2]
```diff
@@ -123,4 +123,20 @@ In this example, the following occurs:
 See <@links.server id="keycloak-truststore"/> for how
 to configure a Keycloak Truststore so that Keycloak is able to perform outgoing requests using TLS.
 
+== Configure Keycloak Server with IPv4 or IPv6
```
[the remark this answers is not in the document]
Comment (mabartos) [reply docs/guides/server/outgoinghttp.adoc:126]: You're right. The section is not very suitable for this.

**Comment by mabartos** [comment]
The most relevant section for including the settings for IPv4 and IPv6 is probably the Production configuration; doesn't have to be changed.

**Review by vmuzikar — APPROVED** [review]

**Review by vmuzikar — CHANGES_REQUESTED** [review]
@mabartos Actually, I realized we should probably mention this in the Migration Guide (we're changing the default behaviour here and it might be surprising to users that KC suddenly listens on IPv6). Could you please add it? Sorry about the last minute note.

**Comment by mabartos** [comment]
@vmuzikar Good point. We could attach some notes to the migration guide to be more transparent in our actions.

**Review by vmuzikar — APPROVED** [review]

**Pull request merged**